### ISSUE TITLE
testing Pkg fails.

### ISSUE THREAD

**Issue by TRC2020** [nep-pack/NonlinearEigenproblems.jl#239]
Hello,

thank you for this awesome pkg. I installed it via 
`(v1.3) pkg> add NonlinearEigenproblems`

and got the following error
```
(v1.3) pkg> test NonlinearEigenproblems
   Testing NonlinearEigenproblems
ERROR: cannot add package `NonlinearEigenproblems = "067cfa3b-fa88-53b2-a873-5b23b3a16e31"` since package `NonlinearEigenproblems = "67991c06-a66a-11e8-39d6-6f5c0e02ffbd"` already exists as a direct dependency.

```

Best Greetings

**Comment by jarlebring** [comment]
Thanks for the report. It's a bit strange. It works on julia 1.1. julia 1.2 gives the  "already exists as a direct dependency" error. 
```
(v1.1) pkg> add NonlinearEigenproblems
```
...
```
(v1.1) pkg> test NonlinearEigenproblems
   Testing NonlinearEigenproblems
    Status `/tmp/tmpHf1eHk/Manifest.toml`
```
We should look into changes in the julia test system.

**Comment by eringh** [comment]
You can indeed still test it manually.

The problem seems to come from that the need to add a subproject for testing, in order to exclude unnecessary dependencies from the main package. We hacked that feature some years ago and now something like that seems to be [included in the Julia core in one way since V1.0 and another way on an experimental basis since V1.2](https://julialang.github.io/Pkg.jl/v1.4/creating-packages/#Adding-tests-to-the-package-1). Thus if we want to conform to that we should probably remove the `NonlinearEigenproblemsTest` project and let the `test/Project.toml` and `test/Manifest.toml` be subprojects.

Since the tests can still be run manually I suggest we keep it like this for now and change the test environment when they have converged to a stable way.

**Comment by jarlebring** [comment]
The new version v1.0.3 has a complete update of the versioning and also the testing system. The tests will only run on julia 1.7. This should now be okay. I will close this issue. If the problem is still there or reappears feel free to reopen it.
